src/spi.rs

```diff
@@ -1273,7 +1273,7 @@ macro_rules! spi {
                                 words[i - FIFO_WORDS] = read_value;
 
                             // Finish emptying the read FIFO
-                            } else if i < len + FIFO_WORDS {
+                            } else {
```

Comment:
Ah yes, it's redundant but was done on purpose to help the reader understand the code. Fair to remove it though